pkg/edgeview/src/system.go

```diff
@@ -0,0 +1,766 @@
+		s := configMap.GlobalSettings[k]
+		s1 := types.ConfigItemValue{
+			ItemType: s.ItemType,
+			IntValue: s.IntDefault,
+			StrValue: s.StringDefault,
+			BoolValue: s.BoolDefault,
+			TriStateValue: s.TriStateDefault,
+		}
+		if getCfgValue(g) == getCfgValue(s1) {
+			buff := fmt.Sprintf(" %s: %s\n", k, getCfgValue(g))
+			fmt.Printf("%s", buff)
+		} else {
+			buff := fmt.Sprintf(" %s: %s; default %s\n", k, getCfgValue(g), getCfgValue(s1))
+			printColor(buff, colorYELLOW)
+		}
+		//fmt.Printf("   default: %v\n", configMap.GlobalSettings[k])
+	}
+
+	// agent
+	printColor("\n - agent settings:", colorCYAN)
+	for k, g := range configitems.AgentSettings {
+		printColor("  "+k+":  ", colorRED)
+		for k1, g1 := range g {
+			fmt.Printf("    %s, %s\n", k1, getCfgValue(g1))
+		}
+	}
+}
+
+func getConfigItems() types.ConfigItemValueMap {
+	var cfgItem types.ConfigItemValueMap
+	retStr, err := runCmd("cat /persist/status/zedagent/ConfigItemValueMap/global.json", false, false)
+	if err != nil {
+		return cfgItem
+	}
+	_ = json.Unmarshal([]byte(retStr), &cfgItem)
+	return cfgItem
+}
+
+func getCfgValue(g types.ConfigItemValue) string {
+	value := ""
+	switch g.ItemType {
+	case types.ConfigItemTypeInt:
+		value = strconv.Itoa(int(g.IntValue))
+	case types.ConfigItemTypeBool:
+		value = strconv.FormatBool(g.BoolValue)
+	case types.ConfigItemTypeString:
+		value = g.StrValue
+	case types.ConfigItemTypeTriState:
+		if g.TriStateValue == types.TS_NONE {
+			value = "None"
+		} else if g.TriStateValue == types.TS_DISABLED {
+			value = "Disabled"
+		} else if g.TriStateValue == types.TS_ENABLED {
+			value = "Enabled"
+		} else {
+			value = "un-supported"
+		}
+	default:
+		value = "un-supported"
+	}
+	return value
+}
+
+func getDownload() {
+	pubsubSvs("/run/", "volumemgr", "DownloaderConfig")
+	pubsubSvs("/run/", "downloader", "DownloaderStatus")
+
+	getMetricsMap("/run/downloader/MetricsMap/", nil, true)
+	checkDownload("/persist/downloads")
+	checkDownload("/persist/vault/downloader")
+	checkDownload("/persist/vault/verifier")
+}
+
+func runPS(opt string) {
+	var item string
+	if strings.Contains(opt, "ps/") {
+		opts := strings.SplitN(opt, "ps/", 2)
+		item = opts[1]
+	}
+	_, _ = runCmd("ps aux | grep "+item, false, true)
+}
+
+func checkDownload(dir string) {
+	retStr, err := runCmd("ls -lR "+dir+" | grep root | grep -v drwx | grep -v dr-x | awk '{print $9}'", false, false)
+	if err != nil || len(retStr) == 0 {
+		return
+	}
+	printColor(" - in "+dir, colorCYAN)
+	lines := splitLines(retStr)
+	for _, jfile := range lines {
+		retStr, err := runCmd("cd "+dir+" && find . -name "+jfile+" -print", false, false)
+		if err != nil {
+			continue
+		}
+		paths := strings.SplitN(retStr, "./", 2)
+		if len(paths) < 2 {
+			continue
+		}
+		path := paths[1]
+		retStr, err = runCmd("ls -l "+dir+"/"+path, false, false)
+		if err == nil {
+			fmt.Printf("%s\n", retStr)
+		}
+	}
+}
+
+func runShell(opt string) {
+	shell := strings.SplitN(opt, "shell/", 2)
```

Comment:
How often have you used this against EVE-OS? Can we drop this generic support and instead gather the list of commands which folks want and make them be specific commands?